### DISCUSSION TITLE
How to improve Lottie loading performance (load 23 Lotties in a collection stack views)

### DISCUSSION THREAD

**Discussion by benjaminsenechal** [airbnb/lottie-ios#1630] [General]
Hello,

I faced a several problem to load a collection of Lottie animations. I explain my code. I have a storyboard which contains 24 views. In my view controller, It's referenced as a collection of IBoutlet with AnimationView as type like this `@IBOutlet var lottieCollectionViews: [AnimationView]!`

I have to load one Lottie (json of 80ko) per view, so I configured my code like this :

```
DispatchQueue.global(qos: .userInteractive).async {
            for n in 0...23 {
                let animation = Animation.named("box"+(n+1).description)
                DispatchQueue.main.async {
                    self.lottieCollectionViews[n].animation = animation
                }
            }
            
        }
```

But, I have memory issue (crash) with older iPhone (7 or SE) and it takes around 5/7sec to load. How can I improve the loading?

Thank you so much.

**Comment by calda** [comment]
This approach in general looks good to me -- moving animation parsing off of the main thread will help keep the app responsive.

I see you're using `DispatchQueue.global(qos: .userInteractive)`. The user-interactive QOS is rather high and may be blocking the main thread. Using `DispatchQueue.global(qos: .default)` (which has a lower priority) may make more sense here.
